core/trino-main/src/main/java/io/trino/sql/planner/plan/AggregationNodeBuilder.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.plan;
+
+import io.trino.sql.planner.Symbol;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class AggregationNodeBuilder
+{
+    private final AggregationNode node;
+    @Nullable
```

Comment:
don't use `@Nullable`, use `Optional`

Reply:
Why? I see we use nullable fields (albeit without @Nullable) in other builders see for example `io.trino.metadata.FunctionMetadata.Builder` or `io.trino.metadata.Signature.Builder`

Reply:
anyway, I dropped @Nullable since I'm copying the fields of original `AggregationNode` now